### ISSUE TITLE
Indexer does not accept login without Captcha

### ISSUE THREAD

**Issue by raphaelmaria** [Jackett/Jackett#12756]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Using HTTP Client: HttpWebClient2
Using FlareSolverr: No
Using proxy: Disabled
App config/log directory: /home/raphaelmaria/.config/Jackett
ThreadPool MaxThreads: 32767 workerThreads, 1000 completionPortThreads
Running in Docker: No
File /etc/issue: Ubuntu 20.04.3 LTS \n \l
Jackett variant: CoreLinuxAmdx64
OS version: Unix 5.11.0.43 (64bit OS) (64bit process)
Environment version: 6.0.1 (/opt/Jackett/)
Starting Jackett v0.20.216
```

### Description

On today's date, some of my indexes started asking for revalidation, asking for a captcha, but in the window that opens next, it doesn't edit at all and doesn't answer the open fields, is there any paleative for that?

### Logged Error Messages

Jackett.Common.ExceptionWithConfigData: <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="pt_BR" lang="pt_BR">
<head>
	<title>Login :: BJ-Share :: Semeando Amizades</title>
	<meta name="robots" content="noindex,nofollow">
	<meta http-equiv="X-UA-Compatible" content="chrome=1; IE=edge" />
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
	<link rel="shortcut icon" href="favicon.ico" />
	<link href="static/styles/public/style.css?v=1640813253" rel="stylesheet" type="text/css" />
	<script src="static/functions/jquery.js"></script>
	<script src="static/functions/script_start.js?v=1619390454"></script>
	<script src="static/functions/ajax.class.js?v=1599349412"></script>
	<script src="static/functions/global.js?v=1630449800"></script>
</head>
<body>
    <div align="center">
            <a href="login.php"><div id="logo"></div></a>
<span id="no-cookies" class="hidden warning">Seus cookies estão desativados.<br/><br/></span>
<noscript><span class="warning">BJ-Share precisa do JavaScript para funcionar. Ative o JavaScript no seu navegador.</span><br/><br/>
</noscript>
    <form class="auth_form" name="xalogin" id="xaloginform" method="post" action="login.php" autocomplete="off">
        <div id="msgsPublic">
                            <span class="warning">Você precisa validar o Recaptcha, tente novamente.<br/><br/></span>
        </div>
        <div id="publicFormDiv">
            <input type="text" maxlength="25" name="username" id="username" autofocus="autofocus" placeholder="Usuário" />
            <input type="password" name="password" id="password" placeholder="Senha" />

                        <br />
            <input type="checkbox" id="keeplogged" name="keeplogged" value="1" checked="checked" />
            <label for="keeplogged">Lembrar-me</label><br/><br/>
    <script src="https://www.google.com/recaptcha/api.js" async defer></script>
    <div class="g-recaptcha" data-sitekey="6LcDdqsZAAAAAASBVUYO3WoJ2pX2cuzqJwemkJoC" data-theme="dark"></div>
    <br />
            <input type="submit" value="Login" class="submit"/>
        </div>
    </form>
                <div id="bottomTexts">
            <br><a href="login.php?act=recover" class="tooltip" title="Recuperar Senha">Recuperar senha</a>        </div>
<script>
    cookie.set('cookie_test', 1, 1);
    if (cookie.get('cookie_test') != null) {
        cookie.del('cookie_test');
    } else {
        $('#no-cookies').gshow();
    }
</script>
<div>
    <span><a href="https://bj-share.info/">BJ-Share &copy; 2016 - 2021</a></span>
</div>
</div>
</body>
</html>

   at Jackett.Common.Indexers.BJShare.<>c__DisplayClass17_0.<ApplyConfiguration>b__0() in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BJShare.cs:line 137
   at Jackett.Common.Indexers.BaseIndexer.ConfigureIfOK(String cookies, Boolean isLoggedin, Func`1 onError) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BaseIndexer.cs:line 258
   at Jackett.Common.Indexers.BJShare.ApplyConfiguration(JToken configJson) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BJShare.cs:line 133
   at Jackett.Server.Controllers.IndexerApiController.UpdateConfig(ConfigItem[] config) in /home/vsts/work/1/s/src/Jackett.Server/Controllers/IndexerApiController.cs:line 97
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in /home/vsts/work/1/s/src/Jackett.Server/Middleware/CustomExceptionHandler.cs:line 26

### Screenshots

![Screen Shot 2021-12-31 at 00 25 05](https://user-images.githubusercontent.com/26657697/147802462-64a94edc-70e2-4b37-bb99-29e82fc3137e.png)

**Comment by garfield69** [comment]
It looks like bjshare have added a ReCaptcha challenge to their login page.
Since this indexer was written in C# it will need a C# dev volunteer to rewrite the login to use the cookie method instead of the usual access via login.php
Volunteers welcome!

**Comment by ilike2burnthing** [comment]
@christianfds or @uilton-oliveira see this instead of #12746

**Comment by uilton-oliveira** [comment]
i'll try to take a look at it later today

**Comment by uilton-oliveira** [comment]
PR Created with the fix

**Comment by garfield69** [comment]
v0.20.222

**Comment by jaum20** [comment]
Can someone show me how the cookie looks like? I'm geting the error: "An error occurred while updating this indexer
Your cookie did not work: The user is not logged in. It is possible that the cookie has expired or you made a mistake when copying it. Please check the settings."

the structure I'm using is:

__cfbbid=xxxxxxxxxx; PHPSESSID=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx; session=xxxxxxxx...xxxx==

**Comment by uilton-oliveira** [comment]
@jaum20 it's the right cookie, but it's currently URL DECODED, and it should be URL ENCODED, for example, should not end with == but instead with %3D%3D (there's also others characters in the middle of value that should not be decoded)

Not sure if you decoded it by yourself or your browser did it automatically, but you can place the value of session= in https://www.urlencoder.org/ or try extract it from your browser without decoding it...

Also, make sure to not log out in BJ-Share site, or it will invalidade the cookie...

**Comment by ilike2burnthing** [comment]
This happens if you right-click the cookie and select `Copy value`. What you need to do instead is highlight the cookie text, right-click and select `Copy`. Not sure why the former is decoded and the latter encoded, but it's a pain if you don't know about it and can't work out why your cookie isn't working.

**Comment by raphaelmaria** [comment]
Grato por todos.
Através de uma das dicas com orientação para ativação via Cookie, o acesso foi restabelecido, use o Mozilla Firefox que veio junto com uma distro do meu Ubuntu, que uso em junto ao Jackett, Sonarr, Radarr e Emby.

Tudo voltou a operar.
-

Grateful for everyone.
Through one of the tips with guidance for activation via Cookie, access was reestablished, use Mozilla Firefox that came with a distro from my Ubuntu, which I use in conjunction with Jackett, Sonarr, Radarr and Emby.

Everything went back into operation.

**Comment by jaum20** [comment]
Thank you @uilton-oliveira and @ilike2burnthing . It worked once, but some minutes after I got a lot of errors on Raddar/Sonnar: "Indexers unavailable due to failures for more than 6 hours: Bj-Share". I can't get it to work for more than some minutes :(

**Comment by uilton-oliveira** [comment]
@jaum20 i'm using it myself daily, and it's working fine.. make sure you're not logging out on the website, as it will invalidate the cookie...

**Comment by uilton-oliveira** [comment]
@jaum20 also, try not leaving the site open on browser for long time while inactive... i noticed that it will cause log out for inactivity and will need to login again and reconfigure the cookies...

**Comment by jaum20** [comment]
@uilton-oliveira I just realized that every time I access the site I need to login again, even checking the remember me box, and this is invalidating my cookie and I have to do the whole process again

**Comment by uilton-oliveira** [comment]
@jaum20 something that i did is that i generated the cookie in a browser that i usually don't use (chrome), so if i logout or lose the session in the browser that i currently use, it will not affect the other cookie (BJ-Share allow multiple cookies per user)

**Comment by jaum20** [comment]
@uilton-oliveira I found the real problem. I have a dynamic IP and my ISP makes it change every hour or so, and this makes the site logs me out

**Comment by dexterwick** [comment]
@jaum20 how you solve? Same problem here, dynamic IP.

**Comment by uilton-oliveira** [comment]
@jaum20 @dexterwick try enabling the MFA and see if it help:
![image](https://user-images.githubusercontent.com/7059651/148606587-f6381ee5-ee80-458f-a7a8-aada3ed3f46a.png)

Some peoples on the tracker forum said that it helped.

**Comment by jaum20** [comment]
@uilton-oliveira enabling the MFA worked like a charm
